### ISSUE TITLE
TCK multipart not set as request parameter

### ISSUE THREAD

**Issue by janbartel** [eclipse/jetty.project#9066]
**jetty-12**

According to the spec, **pg 28 Section 3.2 File Upload**:

> For parts with form-data as the Content-Disposition, but without a filename, the string value of the
> part will also be available through the getParameter and getParameterValues methods on
> HttpServletRequest, using the name of the part.

The TCK tests that such a part is available as a parameter value by sending `Content-Disposition: form-data; name="xyz"` with content `1234567abcdefg`, and in the `TestServlet` doing:

```
   response.setContentType("text/html");

    // Do getParameter first, to test if it works if getParts not called.
    out.write("getParameter(\"xyz\"): " + request.getParameter("xyz"));
    out.write("\n\n");
```

We fail this test because we do not parse the parts until `Request.getPart(String)` is called.

Moreover, once we _do_ parse the parts, we do _not_ set the request parameter.

See https://github.com/jakartaee/platform-tck/blob/master/src/com/sun/ts/tests/servlet/api/jakarta_servlet_http/part/URLClient.java#L85
and https://github.com/jakartaee/platform-tck/blob/master/src/com/sun/ts/tests/servlet/api/jakarta_servlet_http/part/TestServlet.java#L42

**Comment by gregw** [comment]
When we test this, can we make sure any fix works with and without the DelayedHandler

**Comment by joakime** [comment]
There was a question on the `jakartaee/servlet` that seems related ...

* https://github.com/jakartaee/servlet/issues/431

**Comment by lachlan-roberts** [comment]
fixed as part of PR #9287
